docs/advanced-features/instrumentation.md

```diff
@@ -0,0 +1,37 @@
+---
+description: Learn how instrument your Next.js app.
+---
+
+> **Note**: This feature is experimental, you need to explicitly opt-in by providing `experimental.instrumentationHook = true;` in your `next.config.js`.
+
+# `instrumentation.ts`
+
+You can export `register` function from this file and we will be called whenever we bootstrap a new server instance.
+When deployed it will be called continuously on each coldboot (but exactly once in each environment).
+
+Many people used imports with side-effects before we released this feature.
+The behavior is similar, but using imports with side-effects is hard to work with and has many down-sides.
+
+So instead of adding `import 'package-with-side-effect'`, you can add this:
```

Comment:
```suggestion
**We recommend importing the `init` function from `package-init` instead of importing packages with side effects**. You can then call it from your custom `register` function defined in `instrumentation.ts`, as demonstrated in the example below:
```